### PR TITLE
Redundant predicate fix and some sql_test refactor

### PR DESCRIPTION
**Pull request by malin1993ml**
This PR fix the redundant predicate terms in index_scan_plan when some of the scanning condition can be satisfied by index search.

E.g.:
Let's say we have one table: "create table foo(id int primary key, value, int);".
Then "select * from foo where id = ?" doesn't need to check any predicate after primary key index search.

This PR also refactored the SQLTestsUtil a little bit to make it easier to check the execution results.

### CODE REVIEW

File: test/include/sql/sql_tests_util.h [hunk 1]
```diff
@@ -66,6 +66,19 @@ class SQLTestsUtil {
   // A another simpler wrapper around ExecuteSQLQuery
   static Result ExecuteSQLQuery(const std::string query);
 
+  // Get the return value of one column as string at a given position
+  // NOTE: Result columns across different rows are unfolded into a single
+  // vector (vector<ResultType>).
+  static std::string GetResultValueAsString(
+      const std::vector<ResultType> &result, size_t idx) {
+    std::string value;
+    for (auto character : result[idx].second) {
```
Comment (apavlo) [inline test/include/sql/sql_tests_util.h]: You can do the same thing with just this:

```
 std::string value(result[idx].second.begin(), result[idx].second.end());
```

File: src/include/expression/expression_util.h [hunk 2]
```diff
@@ -101,6 +101,87 @@ class ExpressionUtil {
...
+   * searchable. That means there are only "and" conjunctions.
+   */
+  static AbstractExpression *RemoveTermsWithIndexedColumns(
+      AbstractExpression *expression, std::shared_ptr<index::Index> index) {
+    LOG_TRACE("Expression Type --> %s",
+              ExpressionTypeToString(expression->GetExpressionType()).c_str());
+
+    size_t children_size = expression->GetChildrenSize();
+    // We shouldn't have a expression that has more than two children
+    PL_ASSERT(children_size <= 2);
```
Comment (apavlo) [inline src/include/expression/expression_util.h]: Ask @MattPerron about this. I think that you can have more than two children for some functions.

File: src/include/expression/expression_util.h [hunk 3]
```diff
@@ -101,6 +101,87 @@ class ExpressionUtil {
...
   }
 
+  /*
+   * This function removes all the terms related to indexed columns within an
+   * expression.
+   *
+   * NOTE: This should only be called after we check that the predicate is index
+   * searchable. That means there are only "and" conjunctions.
+   */
+  static AbstractExpression *RemoveTermsWithIndexedColumns(
```
Comment (apavlo) [inline src/include/expression/expression_util.h]: Thank you for putting this in `ExpressionUtil` and not in `SimpleOptimizer`

File: test/include/sql/sql_tests_util.h (hunk 1 above)
Comment (malin1993ml) [inline test/include/sql/sql_tests_util.h]: Thanks for pointing out. That's an interesting way.

File: src/include/expression/expression_util.h (hunk 2 above)
Comment (malin1993ml) [inline src/include/expression/expression_util.h]: I've checked with Matt. You're right and I've fixed that.

**Pull request merged**